language/move-lang/src/hlir/translate.rs

```diff
@@ -1378,18 +1377,39 @@ fn check_unused_locals(
     locals: &mut UniqueMap<Var, H::SingleType>,
     used: BTreeSet<Var>,
 ) -> BTreeSet<Var> {
+    let signature = context
+        .signature
+        .as_ref()
+        .expect("Signature should always be defined when checking  a function body");
```

Comment:
Nit, slap an "ICE" at the front of that message. 

I try to make sure I put them there at the beginning of any assert so that its clear to a user that its not some sort of super error message.